web/src/topic_list_data.js

```diff
@@ -106,12 +113,25 @@ export function get_list_info(stream_id, zoomed, search_term) {
             unread: num_unread,
             is_zero: num_unread === 0,
             is_muted: is_topic_muted,
+            is_unmuted: is_topic_unmuted,
             is_active_topic,
             url: hash_util.by_stream_topic_url(stream_id, topic_name),
             contains_unread_mention,
         };
 
-        items.push(topic_info);
+        if (!zoomed) {
+            if (is_topic_unmuted) {
+                unmuted_topics.push(topic_info);
+            } else {
+                normal_topics.push(topic_info);
+            }
```

Comment:
It looks like this changes the sort order, not just whether or not the topic is displayed; is that accurate? 

I think that behavior is wrong is the stream isn't muted, so you need a check for whether the stream is muted in this block somehow -- an unmuted topic in a stream that is not muted should be a noop, not reorder to the top of the left sidebar for the stream.

Reply:
> It looks like this changes the sort order, not just whether or not the topic is displayed; is that accurate?

Yes this part of the code only sorts the order but the `should_show_topic()` function decides whether or not the topic is displayed to which I have added the condition to ensure that unmuted topics are displayed by checking is_topic_unmuted that returns true.

> I think that behavior is wrong is the stream isn't muted, so you need a check for whether the stream is muted in this block somehow -- an unmuted topic in a stream that is not muted should be a noop, not reorder to the top of the left sidebar for the stream.

I have updated the PR to ensure that the sorting only occurs when the stream is muted. Although there is currently no option to unmute a topic in an unmuted stream, I have added an if condition to ensure that the sorting is only done in muted streams. This condition ensures that topics are not sorted in an unmuted stream, even if there are some unmuted topics in any case.

Reply:
Yeah the order of operations to get into that state would be first unmuting the topic, and then unmuting the stream.